EventListener/UserContextSubscriber.php

```diff
@@ -87,11 +87,13 @@ public function onKernelRequest(GetResponseEvent $event)
             return;
         }
 
+        $hash = $this->hashGenerator->generateHash();
+        $event->getRequest()->attributes->set($this->hashHeader, $hash);
```

Comment:
I think it's safe to have this under $this->hash, as there is no reason a hash will be changed during a sub request.